packages/serve/src/lib/server.ts

```diff
@@ -42,15 +52,42 @@ export class VulcanServer {
     // Create application
     const app = this.container.get<VulcanApplication>(TYPES.VulcanApplication);
     await app.useMiddleware();
-    await app.buildRoutes(schemas, this.config.types);
+    await app.buildRoutes(schemas, this.config['types']);
     // Run server
-    const server = http.createServer(app.getHandler()).listen(port);
-    this.server = server;
-    return server;
+    this.servers = this.runServer(app, port, httpsPort);
+    return this.servers;
   }
-
   public async close() {
-    if (this.server) this.server.close();
+    if (this.servers) {
+      if (this.servers['http']) this.servers['http'].close();
+      if (this.servers['https']) this.servers['https'].close();
+      this.servers = undefined;
+    }
     this.container.unload();
   }
+
+  /**
+   * Run server for https when config has setup ssl and middleware 'enforce-https' enabled with LOCAL type, or keep http
+   */
+  private runServer(app: VulcanApplication, port: number, httpsPort: number) {
+    const options = getEnforceHttpsOptions(this.config['enforce-https']);
+    if (options && this.config.ssl) {
```

Comment:
Thanks @oscar60310 for reviewing and suggestion, but I could not understand why we need also create https server when `enforcec-https` options is disabled ?
In general, we usually open http server in the local machine, if user would like to create https server, it could use the reverse proxy ?

If you would like to create local https server, it should set the `LOCAL` undertans the `enforce-https` options.

Reply:
Hi @oscar60310 thanks for discussing with me, after discussion, I have made the HTTPS server open when `enforce-https` options use defualt value ( default is `LOCAL type with `enabled` ).

Besides user setup the disable for `enforce-https` options, or the http server and https server will run both.